hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java

```diff
@@ -269,36 +270,145 @@ public void start() {
+   *   - Om Snapshot number - 100000
+   *   - Om Delta snapshot number - 0
+   *   - All Om Tasks snapshot number - 100000
+   *
+   * Case 5: This case will force Recon to run reprocess of all OM tasks.
+   * full snapshot: DB Updated, Tasks reprocessed, delta DB updates also applied, recon restarted or crash,
+   *                but all delta tasks not processed.
+   *   - Om Snapshot number - 100000
+   *   - Om Delta snapshot number - 100010
+   *   - All Om Tasks snapshot number - 100000
+   *
+   * Case 6: This case will force Recon to run reprocess of only those OM tasks whose
+   *         last updated sequence number is less than Om Delta snapshot number.
+   * full snapshot: DB Updated, Tasks reprocessed, delta DB updates also applied, recon restarted or crash,
+   *                but delta tasks not processed.
+   *   - Om Snapshot number - 100000
+   *   - Om Delta snapshot number - 100010
+   *   - Few Om Tasks snapshot number - 100000  , Remaining Om Tasks snapshot number - 100010
+   *
+   * @param fullSnapshotTaskStatusUpdater
+   * @param deltaTaskStatusUpdater
+   * @return reconOmTaskMap
+   */
+  private Map<String, ReconOmTask> verifyAndGetReconOmTaskMapForReprocess(
+      ReconTaskStatusUpdater fullSnapshotTaskStatusUpdater,
+      ReconTaskStatusUpdater deltaTaskStatusUpdater) {
+
+    Map<String, ReconOmTask> reconOmTaskMap = new HashMap<>();
+
+    for (Map.Entry<String, ReconOmTask> entry : reconTaskController.getRegisteredTasks().entrySet()) {
+      String taskName = entry.getKey();
+      ReconOmTask task = entry.getValue();
+      ReconTaskStatusUpdater taskStatusUpdater = taskStatusUpdaterManager.getTaskStatusUpdater(taskName);
+
+      if (taskStatusUpdater == null) {
+        continue; // Handle null cases
+      }
+
+      boolean isBootstrapTask =
+          shouldProcessTaskForBootstrap(fullSnapshotTaskStatusUpdater, deltaTaskStatusUpdater, taskName,
+              taskStatusUpdater);
+      boolean isDeltaTask =
+          shouldProcessTaskForDelta(fullSnapshotTaskStatusUpdater, deltaTaskStatusUpdater, taskName, taskStatusUpdater);
+
+      if (isBootstrapTask || isDeltaTask) {
+        if (reconOmTaskMap.containsKey(taskName)) {
+          LOG.warn("Duplicate task detected in reconOmTaskMap: {}", taskName);
+        }
+        reconOmTaskMap.put(taskName, task);
+      }
+    }
+
     if (!reconOmTaskMap.isEmpty()) {
-      LOG.info("Task name and last updated sequence number of tasks, that are not matching with " +
-          "the last updated sequence number of OmDeltaRequest task:\n");
+      LOG.info("Tasks with mismatched last updated sequence numbers:");
       LOG.info("{} -> {}", deltaTaskStatusUpdater.getTaskName(), deltaTaskStatusUpdater.getLastUpdatedSeqNumber());
-      reconOmTaskMap.keySet()
-          .forEach(taskName -> {
-            LOG.info("{} -> {}", taskName,
-                taskStatusUpdaterManager.getTaskStatusUpdater(taskName).getLastUpdatedSeqNumber());
```

Comment:
how fullSnapshotTask, delta task and taskStatus are related?

Reply:
We don't want to get the DB updates again, just want OM tasks to reprocess if they failed in their last run whether it was bootstrap case or delta updates case and Recon was need to be restarted or crashed.